### PR TITLE
[RDBMS] `az mysql flexible-server identity/ad-admin`: User Identity and AAD Admin for MySQL flexible server

### PR DESCRIPTION
**Pull request by alanenriqueo**
**Related command**
User identity operations:
- `az mysql flexible-server identity assign`
- `az mysql flexible-server identity list`
- `az mysql flexible-server identity show`
- `az mysql flexible-server identity remove`

AAD Admin operations:
- `az mysql flexible-server ad-admin create`
- `az mysql flexible-server ad-admin list`
- `az mysql flexible-server ad-admin show`
- `az mysql flexible-server ad-admin delete`

**Description**
We need the ability to add/remove user identities and add/remove Active Directory administrator for MySQL flexible server.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -668,6 +670,30 @@ def _flexible_server_params(command_group):
         with self.argument_context('{} flexible-server backup list'.format(command_group)) as c:
             c.argument('server_name', id_part=None, arg_type=server_name_arg_type)
 
+        # identity
+        if command_group == 'mysql':
+            with self.argument_context('{} flexible-server identity'.format(command_group)) as c:
+                c.argument('server_name', id_part=None, options_list=['--server-name', '-s'], arg_type=server_name_arg_type)
+
+            with self.argument_context('{} flexible-server identity assign'.format(command_group)) as c:
```
Comment (DaeunYim) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:678]: Have you considered making an arg type for this parameter?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py [hunk 2]
```diff
@@ -401,3 +405,213 @@ def flexible_server_version_upgrade(cmd, client, resource_group_name, server_nam
...
+
+    parameters = {
+        'identity': mysql_flexibleservers.models.Identity(
+            user_assigned_identities=identities_map,
+            type="UserAssigned")}
+
+    if isinstance(client, MySqlServersOperations):
+        replica_operations_client = cf_mysql_flexible_replica(cmd.cli_ctx, '_')
+    else:
+        # pending postgres
```
Comment (DaeunYim) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:428]: I was just curious if you know the identity assign logic for PG will use the same flow as mysql (== not need to separate these functions into each mysql and pg files?). 

I'm seeing these differences haven't thought  too much about it thou. lmk what you think 
` parameters = {
        'identity': mysql_flexibleservers.models.Identity(`
Or replicas (I assume PG replica is not ready) 

Properties in identities might also be different.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 3]
```diff
@@ -2015,3 +2026,171 @@ def _test_backups_mgmt(self, database_engine, resource_group, server):
...
+class FlexibleServerIdentityAADAdminMgmtScenarioTest(ScenarioTest):
+    mysql_location = 'northeurope'
+
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(location=mysql_location)
+    def test_mysql_flexible_server_identity_aad_admin_mgmt(self, resource_group):
+        self._test_identity_aad_admin_mgmt('mysql', resource_group)
+
+    def _test_identity_aad_admin_mgmt(self, database_engine, resource_group):
+        login = 'alanenriqueo@microsoft.com'
```
Comment (DaeunYim) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:2040]: Not sure if we are allowed to add this info. Will need CLI team's eye on it

**Review by DaeunYim — COMMENTED** [review]
Overall, looks good to me.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 4]
```diff
@@ -2015,3 +2026,171 @@ def _test_backups_mgmt(self, database_engine, resource_group, server):
...
+        sid = '894ef8da-7971-4f68-972c-f561441eb329'
+
```
Comment (DaeunYim) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:2042]: Not sure if it's okay to add this here. Need second eye from CLI team for this

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 1 above)
Comment (alanenriqueo) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:685]: Sounds good, done.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py (hunk 2 above)
Comment (alanenriqueo) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:428]: Logic flow is the same but maybe the structure of the objects will be different, for both identity and ad-admin. For now, I think I'll leave these functions here in flexible_server_custom_common.py and when I need to implement them for PG, I'll check how similar the objets are: if they differ too much, I'll separate the functions in their respective files.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py (hunk 4 above)
Comment (alanenriqueo) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:2042]: There is a [similar test](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_commands.py#L850) for single server.

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:2042]: My suggestion is to create a test user like: https://github.com/Azure/azure-cli/blob/79d3e4d4cab6830ecd6cb4a142895a353f3a9391/src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py#L870

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**